base_layer/core/src/base_node/sync/block_sync/synchronizer.rs

```diff
@@ -271,15 +288,16 @@ impl<B: BlockchainBackend + 'static> BlockSynchronizer<B> {
             let timestamp = header.timestamp();
 
             if header.header().prev_hash != prev_hash {
-                return Err(BlockSyncError::PeerSentBlockThatDidNotFormAChain {
-                    expected: prev_hash.to_hex(),
-                    got: header.header().prev_hash.to_hex(),
-                });
+                return Err(BlockSyncError::ProtocolViolation(format!(
```

Comment:
I prefer the prev error? Why did you change it?
Having a concrete type for concrete error is better, than having a string encased in an error.